backend/inspirehep/orcid/domain_models.py

```diff
@@ -125,42 +125,56 @@ def push(self):
             putcode = self._post_or_put_work(putcode)
         except orcid_client_exceptions.WorkAlreadyExistsException:
             # We POSTed the record as new work, but it failed because the work
-            # already exists (identified by the external identifiers).
+            # with the same identifier is already in ORCID.
             # This means we do not have the putcode, thus we cache all
-            # author's putcodes and PUT the work again.
-            try:
-                if self.do_fail_if_duplicated_identifier:
+            # author's putcodes. If the putcode wasn't found we try to push
+            # once again works with clashing identifier to update them and
+            # resolve conflict.
```

Comment:
The explanation is not very clear. What about something like

We POSTed the record as new work, but it failed because a work with the same identifier is already in ORCID. This can mean two things:
1. the record itself is already in ORCID, but we don't have the putcode;
2. a different record with the same external identifier is already in ORCID.
We first try to fix 1. by caching all author's putcodes and PUT the work again. If the putcode wasn't found we are probably facing case 2. so we try to push once again works with clashing identifiers to update them and resolve the potential conflict.